eureka2-core/src/main/java/com/netflix/eureka2/connection/RetryableConnectionFactory.java

```diff
@@ -124,5 +129,23 @@ public void call(CHANNEL channel) {
         );
     }
 
-    protected abstract Observable<Void> executeOnChannel(CHANNEL channel, OP op);
```

Comment:
That does not answer the question, how we can now pass channel input. Shall I keep the abstract method or add another function argument. For consistency the latter is better, but the number of arguments becomes too big. We might be force to introduce a builder for more clarity.